### DISCUSSION TITLE
How to do cache busting in production?

### DISCUSSION THREAD

**Discussion by FelixWeis** [leptos-rs/leptos#761] [Q&A]
Leptos functions effectively in the development environment, but for production, we aim to extensively cache WASM, JS, and CSS at the edge. The standard cargo-leptos examples do not demonstrate cache invalidation methods after deploying an updated commit. A potential solution could involve adding a query string, such as `?<short_commit_id>`, to the end of `.wasm`, `.js`, and `.css` files within the `"index.html"`. Any suggestions on this matter?

**Comment by gbj** [comment]
Yep how does appending a query string to the JS/WASM depending on an env var you set like LEPTOS_APP_VERSION sound? This would be a new feature so I'll suggest we should take it over to a GitHub issue so I don't lose track but something like this should be quite easy to implement, if providing a version number via environment variable works for you. 

EDIT: Or if you want to just make a PR, it should be a fairly simple edit to [`html_parts`](https://github.com/leptos-rs/leptos/blob/main/integrations/utils/src/lib.rs) to look for the env var and then add it to the JS load and WASM preload.

**Reply by sebadob** [reply]
https://github.com/leptos-rs/leptos/issues/2005#issuecomment-1868080398

Works perfectly fine for me
